odl/util/phantom.py

```diff
@@ -526,6 +543,150 @@ def rect(x):
     return out.ufunc.minimum(1, out=out)
 
 
+def cuboid(discr_space, begin, end):
+    """Rectangular cuboid.
+
+    Parameters
+    ----------
+    discr_space : `Discretization`
+        Discretized space in which the phantom is supposed to be created
+    begin : array-like or `float` in [0, 1]
+        The lower left corner of the cuboid within the space grid relative
+        to the extend of the grid
+    end : array-like or `float` in [0, 1]
+        The upper right corner of the cuboid within the space grid relative
+        to the extend of the grid
+
+    Returns
+    -------
+    phantom : `LinearSpaceVector`
+        Returns an element in ``discr_space``
+
+    Example
+    -------
+    >>> import odl
+    >>> space = odl.uniform_discr(0, 1, 6, dtype='float32')
+    >>> print(cuboid(space, 0.5, 1))
+    [0.0, 0.0, 0.0, 1.0, 1.0, 1.0]
+    >>> space = odl.uniform_discr([0, 0], [1, 1], [4, 6], dtype='float32')
+    >>> print(cuboid(space, [0.25, 0], [0.75, 0.5]))
+    [
+        [0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+        [1.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+        [1.0, 1.0, 1.0, 0.0, 0.0, 0.0],
+        [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+    ]
+    """
+    ndim = discr_space.ndim
+    shape = discr_space.shape
+
+    if np.isscalar(begin):
+        begin = (begin,) * ndim
+    if np.isscalar(end):
+        end = (end,) * ndim
+
+    # Create phantom
+    phan = np.zeros(shape)
+
+    slice1 = [slice(None)] * ndim
+
+    for nn in range(ndim):
+        start = np.floor(begin[nn] * shape[nn]).astype(int)
```

Comment:
If non-integers are used in slice object IPython raises a warning:
```
In [9]: np.ones(2)[slice(1.0)]
/usr/local/bin/ipython:1: DeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  #!/usr/bin/python
Out[9]: array([ 1.])

```